src/json_deser.c

```diff
@@ -401,3 +414,140 @@ corto_int16 json_deserialize(corto_object o, corto_string s) {
+{
+    char *ptr = strrchr(id, '/');
+    if (ptr) {
+        *name = ptr + 1;
+        *parentName = id;
+        *ptr = '\0';
+    } else {
+        *name = id;
+        *parentName = NULL;
+    }
+}
+
+static corto_object json_declareObjectByParentNameNameTypeName(const char* parentName, const char* name, const char* typeName)
+{
+    corto_object package = NULL;
+    corto_object type = corto_resolve(NULL, (corto_string)typeName);
+    if (!type) {
+        corto_seterr("Cannot find %s", typeName);
+        goto errorTypeNotFound;
+    }
+    if (!corto_instanceof(corto_type_o, type)) {
+        corto_seterr("%s is not a type", typeName);
+        goto errorNotType;
+    }
+    if (parentName) {
+        corto_object parent = corto_resolve(root_o, (corto_string)parentName);
+        if (!parent) {
+            corto_seterr("Cannot find parent package %s", parentName);
+            goto errorNoParent;
+        }
+        package = corto_declareChild(parent, (corto_string)name, type);
+        corto_release(parent);
+    } else {
+        package = corto_declareChild(root_o, (corto_string)name, type);
+    }
+    corto_release(type);
+
+    return package;
+
+errorNoParent:
+errorNotType:
+    corto_release(type);
+errorTypeNotFound:
+    return NULL;
+}
+
+static corto_object json_declareObjectByIdAndType(const char* id, const char* type)
+{
+    char* idbuf = corto_strdup(id);
+    if (!idbuf) {
+        goto errorIdBuf;
+    }
+    char *name = NULL;
+    char *parentName = NULL;
+    json_splitParentAndName(idbuf, &parentName, &name);
+
+    corto_object o = json_declareObjectByParentNameNameTypeName(parentName, name, type);
+    corto_dealloc(idbuf);
+    return o;
+
+errorIdBuf:
+    return NULL;
+}
+
+corto_int16 json_toObject(corto_object* o, corto_string s)
+{
+    JSON_Value* topValue = json_parse_string(s);
+    if (!topValue) {
+        corto_seterr("Error parsing %s", s);
+        goto errorParsePackageJson;
+    }
+
+    JSON_Object* topObject = json_value_get_object(topValue);
+    if (!topObject) {
+        corto_seterr("Top level value must be a JSON object");
+        goto errorTopLevelValueNotObject;
+    }
+
+    const char* typeName = json_object_get_string(topObject, "type");
+    if (!typeName) {
+        corto_seterr("No \"type\" found in top level object");
+        goto errorNoType;
+    }
+
+    const char* id = json_object_get_string(topObject, "id");
+    if (!id) {
+        corto_seterr("No \"id\" string field found");
+        goto errorNoId;
+    }
+
+    JSON_Value* value = json_object_get_value(topObject, "value");
+    if (!value) {
+        corto_seterr("No \"value\" found");
+        goto errorNoValue;
+    }
+
+    corto_object o2 = json_declareObjectByIdAndType(id, typeName);
+    if (!o2) {
+        goto errorDeclare;
+    }
+
+    char* valueStr = json_serialize_to_string(value);
+    if (!valueStr) {
+        corto_seterr("Error serializing value to string");
```

Comment:
I'm not going to comment on more `corto_seterr` calls. In general:
- Take into account that these errors will show up at some top level API call, and you have to make sure enough information is in the error so that the user can figure out what is going wrong
- Never use capitals
- Always put '' around variable parts
- Never use ""